compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2024,7 +2024,7 @@ fn add_order_independent_options(
         } else {
             ""
         };
-        cmd.arg(format!("--dynamic-linker={}ld.so.1", prefix));
+        cmd.arg(format!("-dynamic-linker={}ld.so.1", prefix));
```

Comment:
Can we confirm that this change doesn't affect other platforms negatively?

Reply:
https://github.com/rust-lang/rust/blob/0ea89f4db44a3306d34574509ef56ffc248c4e1f/compiler/rustc_codegen_ssa/src/back/link.rs#L2021

Gates this line